cumulus/zombienet/zombienet-sdk/tests/propagation_time/mod.rs

```diff
@@ -0,0 +1,239 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::time::{Duration, Instant};
+
+use anyhow::anyhow;
+
+use cumulus_zombienet_sdk_helpers::ensure_para_throughput;
+use polkadot_primitives::Id as ParaId;
+use zombienet_sdk::{
+	subxt::{ext::futures, OnlineClient, PolkadotConfig},
+	AddCollatorOptions, LocalFileSystem, Network, NetworkConfigBuilder, NetworkNode,
+};
+
+const PARA_ID: u32 = 2000;
+const BEST_BLOCK_METRIC: &str = "block_height{status=\"best\"}";
+
+#[ignore = "Slow test used to measure block propagation time in a sparsely connected network"]
+#[tokio::test(flavor = "multi_thread")]
+async fn sparsely_connected_network_block_propagation_time() -> Result<(), anyhow::Error> {
+	log::warn!("This test is slow. It will take a long time to complete.");
+	tokio::time::sleep(Duration::from_secs(3)).await;
+
+	let mut num_failures = 0;
+	let mut tests_remaining = 10;
+	let mut propagation_times = Vec::new();
+
+	// Try multiple tests to get a better average.
+	while tests_remaining > 0 {
+		log::info!("Running test with {tests_remaining} tests remaining");
+		if num_failures > 5 {
+			anyhow::bail!("Too many failures ({num_failures}), aborting further tests.");
+		}
+		match run_network().await {
+			Ok(propagation_time) => {
+				log::info!("Propagation time: {propagation_time} seconds");
+				propagation_times.push(propagation_time);
+			},
+			Err(e) => {
+				log::error!("Test failed: {e}");
+				num_failures += 1;
+			},
+		}
+		tests_remaining -= 1;
+	}
+
+	let avg = propagation_times.iter().sum::<f64>() / propagation_times.len() as f64;
+	log::info!("Average propagation time: {avg} seconds");
+	propagation_times.sort_by(|a, b| a.partial_cmp(b).unwrap());
+	let median = if propagation_times.len() % 2 == 0 {
+		(propagation_times[propagation_times.len() / 2 - 1] +
+			propagation_times[propagation_times.len() / 2]) /
+			2.0
+	} else {
+		propagation_times[propagation_times.len() / 2]
+	};
+	log::info!("Median propagation time: {median} seconds");
```

Comment:
We could aslo see p90 / p99. Maybe we'll get some improvements there.
If you want you could use [this](https://docs.rs/statrs/0.18.0/statrs/statistics/trait.OrderStatistics.html) for computing them.